src/node.rs

```diff
@@ -315,6 +319,132 @@ impl PbftNode {
         }
     }
 
+    /// Verifies an individual consensus vote
+    ///
+    /// Returns the signer ID of the wrapped PbftMessage, for use in further verification
+    fn verify_consensus_vote(
+        vote: &PbftSignedCommitVote,
+        seal: &PbftSeal,
+    ) -> Result<Vec<u8>, PbftError> {
+        let message: PbftMessage = protobuf::parse_from_bytes(&vote.get_message_bytes())
+            .map_err(PbftError::SerializationError)?;
+
+        if message.get_block().block_id != seal.previous_id {
+            return Err(PbftError::InternalError(format!(
+                "PbftMessage block ID ({:?}) doesn't match seal's previous id ({:?})!",
+                message.get_block().get_block_id(),
+                seal.previous_id
+            )));
+        }
+
+        let header: ConsensusPeerMessageHeader =
+            protobuf::parse_from_bytes(&vote.get_header_bytes())
+                .map_err(PbftError::SerializationError)?;
+
+        let key = Secp256k1PublicKey::from_hex(&hex::encode(&header.signer_id)).unwrap();
+
+        let context = create_context("secp256k1")
+            .map_err(|err| PbftError::InternalError(format!("Couldn't create context: {}", err)))?;
+
+        match context.verify(
+            &hex::encode(vote.get_header_signature()),
+            vote.get_header_bytes(),
+            &key,
+        ) {
+            Ok(true) => {}
+            Ok(false) => {
+                return Err(PbftError::InternalError(
+                    "Header failed verification!".into(),
+                ))
+            }
+            Err(err) => {
+                return Err(PbftError::InternalError(format!(
+                    "Error while verifying header: {:?}",
+                    err
+                )))
+            }
+        }
+
+        verify_sha512(vote.get_message_bytes(), header.get_content_sha512())?;
+
+        Ok(message.get_info().get_signer_id().to_vec())
+    }
+
+    /// Verifies the consensus seal from the current block, for the previous block
+    fn verify_consensus_seal(
+        &mut self,
+        block: &Block,
+        state: &mut PbftState,
+    ) -> Result<(), PbftError> {
+        // We don't publish a consensus seal until block 2, so we don't verify it
+        // until block 3
+        if block.block_num < 3 {
```

Comment:
I think you need to validate the seal starting at block 2.
Block 0 - Genesis
Block 1 - First PBFT block
Block 2 - Second PBFT block, should contain votes for block 1